### PR TITLE
Godel Robots

### PR DESCRIPTION
**Pull request by Jmeyer1292**
This PR adds the robot configurations for the ABB IRB2400.

The original Godel repo had support for the motoman baked right into the `godel_robot_config` package. As we grew the capabilities of Godel and added many new robots, it quickly became difficult to keep all of the robot configurations inside this package. 

I created a `godel_robots` directory that houses a utility package `godel_robot_resources` as well as robot vendor specific sub folders. These vendor folders contain directories for each supported robot and those in turn include:
1. Descartes configuration
2. Support package (URDFs)
3. MoveIt configuration
4. IKFast plugin

More than almost anything else, this package was changed constantly and I'm very open to improve this way of handling robot configurations. I also have support for the UR5, UR10, Motoman SIA20D, ABB6600, and ABB2600 but some of these require some experimental components, so I'm holding off for now.

### CODE REVIEW

**Comment by VictorLamoine** [comment]
Just a suggestion: Maybe you could centralize your (SwRI) robot definitions, sensors etc. in an external package and depend on it, something like:
https://github.com/InstitutMaupertuis/institut_maupertuis_robots_descriptions

I like this solution because it does not "pollute" the code repository with many robot definitions that are alike.

**Comment by Jmeyer1292** [comment]
@VictorLamoine We have been arguing about doing just that internally. @jrgnicho can weigh in, but I believe the consensus was that until we settle on some interfaces that are semi fixed it will be easier to keep everything in one repo and make one pull-request to change things.

File: godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/launch/moveit_planning_execution.launch [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    <include file="$(find industrial_robot_simulator)/launch/robot_interface_simulator.launch" />
+  </group>
+
+  <!-- run the "real robot" interface nodes -->
+  <!--   - this typically includes: robot_state, motion_interface, and joint_trajectory_action nodes -->
+  <!--   - replace these calls with appropriate robot-specific calls or launch files -->
+  <group unless="$(arg sim)">
+    <include file="$(find abb_irb2400_support)/launch/robot_interface_download_irb2400.launch" >
+      <arg name="robot_ip" value="$(arg robot_ip)"/>
+      <!-- <arg name="controller" value="$(arg controller)"/> -->
```
Comment (Jmeyer1292) [inline godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/launch/moveit_planning_execution.launch]: Old Motoman codes should go

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/config/scan_plan.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+scan_plan:
+  scan_width: 0.01
+  traverse_spd: 0.0025
+  approach_distance: 0.15
+  overlap: 0.001
+  margin: 0
+  quality_metric: "rms"
```
Comment (Jmeyer1292) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/config/scan_plan.yaml:7]: This will need to be fixed as we go forward without string based quality-metrics

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/launch/irb2400_blending.launch [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    <arg name="robot_model_plugin" value="$(arg robot_model_plugin)"/>
+  </include>
+
+  <!-- Process execution nodes: These monitor the state of robot's execution of the planned path. They can be, and in this case are, robot or vendor specific -->
+  <node name="blend_process_execution" pkg="godel_process_execution" type="abb_blend_process_service_node"/>
+  <node name="scan_process_execution" pkg="godel_process_execution" type="keyence_process_service_node"/>
+
+  <!-- Bring up simulated robot that can be visualized under the tf namespace 'simulation'
+       Allows for previewing paths generated before execution. -->
+  <include file="$(find simulator_service)/launch/simulation.launch"/>
```
Comment (Jmeyer1292) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/launch/irb2400_blending.launch:23]: This filename will change

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/urdf/abb_blending_eff_macro.xacro [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    <link name="${prefix}tcp_frame"/>
+
+    <link name="${prefix}keyence_sensor_optical_frame"/>
+    <link name="${prefix}keyence_tcp_frame" />
+
+    <link name="${prefix}gripper_body">
+      <visual>
+        <geometry>
+          <mesh filename="package://godel_robot_config/meshes/visual/blending_eff.stl" />
```
Comment (Jmeyer1292) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/urdf/abb_blending_eff_macro.xacro]: This mesh is complicated and ill-formed, so launching it as collision geometry will make MoveIt choke every time it loads a model. I need to move the model's from this directory into either this repo or the shared resources one.

File: godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/launch/moveit_planning_execution.launch (hunk 1 above)
Comment (jrgnicho) [inline godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/launch/moveit_planning_execution.launch]: Has it already been removed? I don't see any Motoman specific stuff here.

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/config/scan_plan.yaml (hunk 2 above)
Comment (jrgnicho) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/config/scan_plan.yaml:7]: What is broken?

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/urdf/abb_blending_eff_macro.xacro (hunk 4 above)
Comment (jrgnicho) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/urdf/abb_blending_eff_macro.xacro]: I don't follow why moveit will have difficulty loading this model.

File: godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/config/ompl_planning.yaml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    border_fraction: 0.9  # Fraction of time focused on boarder default: 0.9 (0.0,1.]
+    failed_expansion_score_factor: 0.5  # When extending motion fails, scale score by factor. default: 0.5
+    min_valid_path_fraction: 0.5  # Accept partially valid moves above fraction. default: 0.5
+  RRTkConfigDefault:
+    type: geometric::RRT
+    range: 0.0  # Max motion added to tree. ==> maxDistance_ default: 0.0, if 0.0, set on setup()
+    goal_bias: 0.05  # When close to goal select goal, with this probability? default: 0.05
+  RRTConnectkConfigDefault:
+    type: geometric::RRTConnect
+    range: 0.0  # Max motion added to tree. ==> maxDistance_ default: 0.0, if 0.0, set on setup()
```
Comment (jrgnicho) [inline godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/config/ompl_planning.yaml]: Since you are using RRTConnectkConfigDefault for the macro planning stage, it may be beneficial to adjust the "range " parameter to a small number (0.01,0.05,0.1, etc) in order to get well behaved motion more frequently.

File: godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/launch/joystick_control.launch [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    <param name="dev" value="$(arg dev)" /> <!-- Customize this to match the location your joystick is plugged in on-->
+    <param name="deadzone" value="0.2" />
+    <param name="autorepeat_rate" value="40" />
+    <param name="coalesce_interval" value="0.025" />
+  </node>
+
+  <!-- Launch python interface -->
+  <node pkg="moveit_ros_visualization" type="moveit_joy.py" output="screen" name="moveit_joy"/>
+        
+</launch>
```
Comment (jrgnicho) [inline godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/launch/joystick_control.launch:17]: Where in the blending process workflow is this used? Was it automatically generated?

Comment (Jmeyer1292) [inline godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/launch/joystick_control.launch:17]: Yes, it was automatically generated.

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/yaml/godel_mutable_joint_states.yamlnew [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+kinect2_link_x_joint: -0.05490207720545535
+kinect2_link_y_joint: -0.1108059263096891
+kinect2_link_yaw_joint: -1.573233882247353
+kinect2_link_z_joint: -0.01737111616208457
+target_frame_pitch_joint: -0.01787266651478813
+target_frame_roll_joint: -0.01505099432981459
+target_frame_x_joint: 0.8587948355454501
+target_frame_y_joint: 0.09443069397431932
+target_frame_yaw_joint: -1.545810493677323
+target_frame_z_joint: 0.01343273444585487
```
Comment (jrgnicho) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/yaml/godel_mutable_joint_states.yamlnew]: Is this file still needed?

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/yaml/kinect2_camera_def.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+#    ros_camera_lti  listens to the camera_optical_frame relative to reference directly 
+#    ros_camera_housing_lti  listens to the camera_optical_frame relative to reference directly 
+#               must define camera_housing_frame: /camera_frame, but not used
+#    ros_camera_housing_bti  broadcast the reference_frame to camera_housing_frame,
+#               must define camera_housing_frame: /camera_frame
+#    ros_lti  listens to the reference_frame to camera_optical frame, does not invert
+#    ros_bti  broadcasts to the reference_frame to camera_optical frame, does not invert
+#    ros_camera_housing_cti  interfaces to the mutable_joint_state_publisher 
+#           must define camera_housing_frame: /camera_housing_frame
+#           must define camera_mounting_frame: /camera_mounting_frame
```
Comment (jrgnicho) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/yaml/kinect2_camera_def.yaml]: These comments should go at the top of the file.

File: godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/launch/joystick_control.launch (hunk 6 above)
Comment (jrgnicho) [inline godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/launch/joystick_control.launch:17]: OK, that's news to me. you may disregard this question then.

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/config/scan_plan.yaml (hunk 2 above)
Comment (Jmeyer1292) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/config/scan_plan.yaml:7]: Well Gijs thought that denoting the active quality_metric via string, as seen here, was brittle. So I changed it to be an enumeration, but that means I'll need to switch this to a number and probably add a description so that readers know what they're looking at.

File: godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/config/ompl_planning.yaml (hunk 5 above)
Comment (Jmeyer1292) [inline godel_robots/abb/godel_irb2400/godel_irb2400_moveit_config/config/ompl_planning.yaml]: +1

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/yaml/godel_mutable_joint_states.yamlnew (hunk 7 above)
Comment (Jmeyer1292) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/yaml/godel_mutable_joint_states.yamlnew]: +1
No

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/yaml/kinect2_camera_def.yaml (hunk 8 above)
Comment (Jmeyer1292) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/yaml/kinect2_camera_def.yaml]: Sure

File: godel_robots/abb/godel_irb2400/godel_irb2400_support/config/scan_plan.yaml (hunk 2 above)
Comment (jrgnicho) [inline godel_robots/abb/godel_irb2400/godel_irb2400_support/config/scan_plan.yaml:7]: +1

**Comment by jrgnicho** [comment]
+1. You can merge
